backend/controller/door.go

```diff
@@ -67,11 +67,17 @@ func IsDoorOpen(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		return
 	}
-	_, hasAccess := models.DoorUnlocked[dId]
-	if hasAccess {
-		w.WriteHeader(200)
-	} else {
+	uId, hasAccess := models.DoorUnlocked[dId]
+	if !hasAccess {
 		http.Error(w, "Door not unlocked", http.StatusUnauthorized)
+		return
+	}
+
+	err = UpdateLastTimeIn(w, uId)
+	if err != nil {
+		http.Error(w, "Unable to update last time in", http.StatusInternalServerError)
+		return
```

Comment:
I'm confused I thought this was the endpoint that was constantly pinged by the door? Does this logic belong here?

Reply:
it won't ever reach that step, this is when the user unlocks the door so yes why would this not be where the logic belongs? they unlock door on mobile we update their time in, otherwise we don't update it